app/src/main/java/com/wafflestudio/snutt2/views/logged_in/home/TableListViewModel.kt

```diff
@@ -74,12 +99,10 @@ class TableListViewModel @Inject constructor(
         tableRepository.copyTable(tableId)
     }
 
-    suspend fun checkTableDeletable(tableId: String): Boolean {
-        return currentTableRepository.currentTable
+    suspend fun checkTableDeletable(): Boolean {
```

Comment:
```suggestion
    fun checkTableDeletable(tableId: String): Boolean {
        val tableToDelete = tableRepository.tableMap.value[tableId] ?: return false
        return tableRepository.tableMap.value.values.filter {
            it.semester == tableToDelete.semester && it.year == tableToDelete.year
        }.size > 1
    }
```
요런식으로 (구체적인 코드는 입맛대로 ㄱㄱ)